components/cost-management/staging/policies/propogate_namespace_labels.yaml

```
@@ -0,0 +1,55 @@
+---
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: propagate-cost-management-labels
+  annotations:
+    policies.kyverno.io/title: Propagate Cost-Center from Namespace
+    policies.kyverno.io/category: Cost Management
+    policies.kyverno.io/severity: low
+    policies.kyverno.io/subject: Pod
+    policies.kyverno.io/description: >-
+      This policy ensures that: 
+      (1) The `cost-center` label is propagated from non-managed tenant namespaces to pods.
+      (2) The `insights_cost_management_optimizations` label is set to `true` on Pods to which the `cost-center` label was propagated.
```

as we are allowing the policy to fail without preventing the pod from being created, I'd suggest to change this `ensure` with something like `tries to`. 
```suggestion
      This policy tries to that: 
      (1) Propagate the `cost-center` label from non-managed tenant namespaces to pods.
      (2) Set the `insights_cost_management_optimizations` label is set to `true` on Pods to which the `cost-center` label was propagated.
      If the webhook fails, the pod will be created without the labels.
```